### DISCUSSION TITLE
Please have a proper security audit done (by security professionals)

### DISCUSSION THREAD

**Discussion by ThiefMaster** [rustdesk/rustdesk#8392] [General]
I realize that this will cost you some money, but it would greatly give those of us who not only value our privacy (self-hosted is great for that :)) but also our computer's security.

Because right now, while certainly impressive when it comes to functionality this tool is just a random piece of software where nobody knows if and how much to trust it:

- The contributors (or just a single guy?) are hidden behind the `rustdesk` GitHub account (very uncommon, typically people use GitHub accounts that are tied to their person, not to their project/company). There's a reference to a LinkedIn profile but besides being a terrible platform (who, maybe besides some HR people, actually like LinkedIn?!) it also does not show much details even if you pass through the login gate
- Discussions in the Discord channel are very basic, and the ChatGPT bot doesn't make this better. But I guess it's more meant to be a community help place and not a place to get direct responses from a core developer.
- Documentation on security-related features such as the server's `-k` option is kind of messy. When looking at `-k -` (recent server release notes) and `-k _` (most places) it's not clear what's the correct option, and it takes trial+error to figure out that neither of them *require* the clients to have matching keys. For that I need to set `-k <pubkey>`
- Connections and authentication: Are both of the encrypted? Are they *always* encrypted, and if encryption is optional, can it be enforced so an unencrypted connection is dropped before any sensitive data is transmitted? How is the crypto handshake done? Could an MITM gain any useful information, maybe even keyboard input or the password used to authenticate?
- Especially this last part is extremely sensitive - as you can imagine, a serious bug in that code would be "game over". Someone remotely controlling mouse and keyboard is bad enough (but you lock your PC when going AFK, right? :)), but this tool also supports file transfers. That probably doesn't require the desktop to be unlocked.

So please, get a reputable cybersecurity company to do a thorough audit of your product (yes, bot the paid AND the opensource version), or at least convince some white-hat hackers to do some poking to see if there are any vulnerabilities... But considering that you sell a commercial product, I think you should really go for the "full" option and have a security audit done. Obviously I don't mean the kind that CEOs love (ticking all the boxes about processes, playbooks, whatever), but the kind where someone highly technical looks into the code.

**Comment by clemlesne** [comment]
What about a crowdfunding?

**Comment by southerndoc** [comment]
Isn't open source auditable by its nature?
